classes/core/PKPAppKey.php

```diff
@@ -0,0 +1,185 @@
+<?php
+
+/**
+ * @file classes/core/PKPAppKey.php
+ *
+ * Copyright (c) 2014-2024 Simon Fraser University
+ * Copyright (c) 2000-2024 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class PKPAppKey
+ *
+ * @ingroup core
+ *
+ * @brief Class to manage app key related behaviours
+ */
+
+namespace PKP\core;
+
+use Exception;
+use PKP\config\Config;
+use PKP\config\ConfigParser;
+use Illuminate\Support\Str;
+use Illuminate\Encryption\Encrypter;
+
+class PKPAppKey
+{
+    /**
+     * The supported cipher algorithms and their properties.
+     *
+     * @var array
+     */
+    private static array $supportedCiphers = [
+        'aes-128-cbc' => ['size' => 16, 'aead' => false],
+        'aes-256-cbc' => ['size' => 32, 'aead' => false],
+        'aes-128-gcm' => ['size' => 16, 'aead' => true],
+        'aes-256-gcm' => ['size' => 32, 'aead' => true],
+    ];
+
+    /**
+     * The default cipher algorithms
+     *
+     * @var string
+     */
+    private static string $defaultCipher = 'aes-256-cbc';
+
+    /**
+     * Get the list of supported ciphers
+     */
+    public static function getSupportedCiphers(): array
+    {
+        return self::$supportedCiphers;
+    }
+
+    /**
+     * Get the defined cipher
+     */
+    public static function getCipher(): string
+    {
+        return Config::getVar('security', 'cipher', self::$defaultCipher);
+    }
+
+    /**
+     * Has the app key defined in config file
+     */
+    public static function hasKey(): bool
+    {
+        return !empty(Config::getVar('general', 'app_key', ''));
```

Comment:
You can leave the 3rd parameter blank if you don't want to specify a default; also below

Reply:
(minor, but still needs attention)